drivers/tee/optee/core.c

```diff
@@ -570,10 +571,55 @@ static struct platform_driver optee_driver = {
 	.remove = optee_remove,
 };
 
+static void enable_cpu_perf_counters(void *data)
+{
+#ifdef __aarch64__
+	/* Enable EL0 access to PMU counters. */
+	asm volatile("msr pmuserenr_el0, %0" :: "r"((u64)
```

Comment:
This generic arm/arm64 stuff and doesn't really belong in this driver, compare with `<linux/arm-smccc.h>`